### DISCUSSION TITLE
How to copy page templates from a site to another site?

### DISCUSSION THREAD

**Discussion by mohammadamer** [pnp/pnpcore#1250] [General]
I have a mission to copy page templates from a site to another site. I managed to get all files in Templates folder in SitePages library and did copying them by running post request that calls a Power Automate flow. 
The issue is that pages are copied to the templates folder but as pages not page templates.

Is there a way to set the pages (files) to be page templates after copying - either in Power Automate or PnPCore ?

Has anyone encountered such issue? appreciate any help?

**Comment by jansenbe** [comment]
@mohammadamer : to make a page a template you can call the `IPage.GetTemplatesFolderAsync()` method to ensure there's a valid templates folder in the target sitepages library before you copy in the pages into that folder. Once the folder is correctly configured the pages should show up as templates
